python-project-template/.pre-commit-config.yaml.jinja

```diff
@@ -21,7 +21,7 @@ repos:
       - id: pytest-check
         name: pytest-check
         description: Run unit tests with pytest.
-        entry: python -m pytest --cov=./src --cov-report=html
+        entry: bash -c "if python -m pytest --co -qq; then python -m pytest --cov=./src --cov-report=html; fi"
```

Comment:
This pre-commit hook change addresses #114 (at least on my windows machine)

The logic in spoken language is, "If there are tests that are found, then run the tests." While the error output in the ticket made it appear that the code coverage was failing, what was actually the case was that no tests were collected, and a non-0 exit code was produced. This exit code was read by pre-commit and presented as a failure. 

I have manually tested this on my WSL machine with the following three cases:

1.  No tests are present. Output of pre-commit hook = Passed
2. One test that `assert True`s is present. Output of pre-commit hook = Passed
3. One test that `assert False`s is present. Output of pre-commit hook = Failed, prints pytest output.

Reply:
The approach here was inspired by this pre-commit issue: https://github.com/pre-commit/pre-commit/issues/2802

Reply:
Yes, I agree that `.hooks/pre-commit` would be a useful thing to include in the near future for exactly the reasons you called out here. If we run across this issue again, it will be time to include that I think.